TYPO3.Neos/Classes/TYPO3/Neos/Service/Mapping/NodePropertyConverterService.php

```diff
@@ -0,0 +1,248 @@
+        }
+
+        return $properties;
+    }
+
+    /**
+     * @param NodeInterface $node
+     * @param string $propertyName
+     * @return mixed
+     */
+    protected function getPropertyInternal(NodeInterface $node, $propertyName)
+    {
+        if ($propertyName[0] === '_') {
+            $propertyValue = ObjectAccess::getProperty($node, ltrim($propertyName, '_'));
+        } else {
+            $propertyValue = $node->getProperty($propertyName);
+        }
+
+        $dataType = $node->getNodeType()->getPropertyType($propertyName);
+        $convertedValue = $this->convertValue($propertyValue, $dataType);
+
+        if ($convertedValue === null) {
+            $convertedValue = $this->getDefaultValueForProperty($node->getNodeType(), $propertyName);
+        }
+
+        return $convertedValue;
+    }
+
+    /**
+     * @param mixed $propertyValue
+     * @param string $dataType
+     * @return mixed
+     */
+    protected function convertValue($propertyValue, $dataType)
+    {
+        $rawType = TypeHandling::truncateElementType($dataType);
+
+        // This hardcoded handling is to circumvent rewriting PropertyMappers that convert objects. Usually they expect the source to be an object already and break if not.
+        if (!TypeHandling::isSimpleType($rawType) && !is_object($propertyValue)) {
+            return null;
+        }
+
+        if ($rawType === 'array') {
+            $conversionTargetType = 'array<string>';
+        } elseif (TypeHandling::isSimpleType($rawType)) {
+            $conversionTargetType = TypeHandling::normalizeType($rawType);
+        } else {
+            $conversionTargetType = 'array';
+        }
+
+        $propertyMappingConfiguration = $this->createConfiguration($dataType);
+        try {
+            $convertedValue = $this->propertyMapper->convert($propertyValue, $conversionTargetType, $propertyMappingConfiguration);
+        } catch (\TYPO3\Flow\Property\Exception $exception) {
+            $convertedValue = null;
+        }
+
+        if ($convertedValue instanceof \TYPO3\Flow\Error\Error) {
+            $convertedValue = null;
+        }
+
+        return $convertedValue;
+    }
+
+    /**
+     * Tries to find a default value for the given property trying:
+     * 1) The specific property configuration for the given NodeType
+     * 2) The generic configuration for the property type in setings.
+     *
+     * @param NodeType $nodeType
+     * @param string $propertyName
+     * @return mixed
+     */
+    protected function getDefaultValueForProperty(NodeType $nodeType, $propertyName)
+    {
+        $propertyDefault = $nodeType->getConfiguration('properties.' . $propertyName . 'defaultValue');
+
+        if ($propertyDefault === null) {
+            $propertyType = $nodeType->getPropertyType($propertyName);
+            $propertyDefault = ObjectAccess::getPropertyPath($this->typesConfiguration, $propertyType . '.defaultValue');
```

Comment:
this won't cut it, instead it needs to be fetched using `getDefaultValuesForProperties` on the node type class that includes custom logic